components/brave_shields/browser/brave_shields_util.cc

```diff
@@ -231,6 +232,23 @@ bool IsFirstPartyCosmeticFilteringEnabled(HostContentSettingsMap* map,
   return type == ControlType::BLOCK;
 }
 
+bool ShouldDoDebouncing(HostContentSettingsMap* map, const GURL& url) {
+  // Don't debounce if debounce feature is disabled
+  if (!base::FeatureList::IsEnabled(debounce::features::kBraveDebounce))
+    return false;
+
```

Comment:
we should also short-circuit if the url is not http/https. It's not as relevant since per @pes10k this should only apply to top-level navigations, but we still don't want to run into any cases where we are wasting time parsing javascript urls for instance

Reply:
This is handled inside `GetBraveShieldsEnabled`